roles/postgresql_upgrade/tasks/main.yml

```
@@ -0,0 +1,58 @@
+---
+- name: Get version of postgres data directory
+  slurp:
+    path: /var/lib/pgsql/data/PG_VERSION
+  register: pg_version
+
+- when:
+    - ansible_distribution_major_version == '8'
+    - "'10' in pg_version['content']|b64decode"
+  block:
+    - name: Start postgresql
+      service:
+        name: postgresql
+        state: started
+
+    - name: Collect postgres data
+      shell: psql -lt | grep -E "^\s+postgres"
+      become: true
+      become_user: postgres
+      register: postgres_output
+
+    - name: Set collate and ctype facts
+      set_fact:
+        collate: "{{ postgres_output.stdout.split('|')[3] | trim }}"
+        ctype: "{{ postgres_output.stdout.split('|')[4] | trim }}"
+
+    - name: Stop postgresql
+      service:
+        name: postgresql
+        state: stopped
+
+    - name: Update postgresq.conf data directory
```

```suggestion
    - name: Update postgresql.conf data directory
```

Update to what? The task essentially deletes the config line.